editor/src/core/shared/multiplayer.ts

```diff
@@ -0,0 +1,86 @@
+import type { User } from '@liveblocks/client'
+import type { Presence, UserMeta } from '../../../liveblocks.config'
+import { safeIndex, uniqBy } from './array-utils'
+import { getPreferredColorScheme } from '../../uuiui'
+
+export type MultiplayerColor = {
+  background: string
+  foreground: string
+}
+
+export const multiplayerColors = {
+  light: [
+    { background: '#0092C0', foreground: 'white' },
+    { background: '#128400', foreground: 'white' },
+    { background: '#2A00D1', foreground: 'white' },
+    { background: '#5C00D1', foreground: 'white' },
+    { background: '#87C700', foreground: 'black' },
+    { background: '#8C6D00', foreground: 'white' },
+    { background: '#9C0000', foreground: 'white' },
+    { background: '#B16CB7', foreground: 'white' },
+    { background: '#B60479', foreground: 'white' },
+    { background: '#E7B400', foreground: 'black' },
+  ],
+  dark: [
+    { background: '#007DA4', foreground: 'white' },
+    { background: '#19B500', foreground: 'black' },
+    { background: '#3EFFF3', foreground: 'black' },
+    { background: '#6842FF', foreground: 'white' },
+    { background: '#C4FF46', foreground: 'black' },
+    { background: '#CE00D3', foreground: 'white' },
+    { background: '#DD0000', foreground: 'white' },
+    { background: '#FF00A8', foreground: 'white' },
+    { background: '#FFBA08', foreground: 'black' },
+    { background: '#FFFF00', foreground: 'black' },
+  ],
+}
+
+export function multiplayerColorFromIndex(colorIndex: number | null): MultiplayerColor {
+  const fallbackColor = {
+    background: '#000',
+    foreground: '#fff',
+  }
+  if (colorIndex == null) {
+    return fallbackColor
+  }
+
+  const colors =
+    getPreferredColorScheme() === 'dark' ? multiplayerColors.dark : multiplayerColors.light
+  return safeIndex(colors, colorIndex) ?? fallbackColor
+}
+
+/**
+ * Returns the initials from the given name, to be displayed on e.g. avatars or comments.
+ * If the name is made of multiple words, the first letter of the first two words will be returned ('John Doe' -> 'JD').
+ * If the name is made of a single word, the first two letters of the word will be returned ('John' -> 'JO').
+ * If the name is shorter than two letters, the result will be padded with 'X' characters ('F' -> 'FX').
```

Comment:
why not just leave it shorter?

Reply:
I think at least for me the assumption is to have uniform 2-letter initials for now (there are a lot of possible further variations on this, e.g. names with interfixes so I wanted to go with the most stable solution for now